rust/lance-linalg/src/simd/f16.c

```diff
@@ -20,38 +20,38 @@
 #endif // __X86_64__
 
 /// Works on NEON + FP16 or AVX512FP16
-_Float16 norm_l2_f16(const _Float16 *data, uint32_t dimension) {
+float norm_l2_f16(const _Float16 *data, uint32_t dimension) {
   _Float16 sum = 0;
```

Comment:
Does this return float for overflow? It is a problem that the `sum` variable is still `_Float16`?

Reply:
Return f32 is for the consumer to sort / build f32 lookup table, which is usually faster on the chips.

This is less about handling overflow, because if overflow happens, it is already done in the `_Float16 sum`.